pyhik/hikvision.py

```diff
@@ -132,6 +132,77 @@ def current_event_states(self):
         """Return Event states dictionary"""
         return self.event_states
 
+    def get_motion_detection(self):
+        url = '%s/MotionDetection/1' % self.root_url
+        try:
+            response = self.hik_request.get(url)
+        except requests.exceptions.RequestException as err:
+            _LOGGING.error('Unable to fetch MotionDetection, error: %s', err)
+            self.motion_detection = None
+            return self.motion_detection
+
+        if response.status_code == requests.codes.unauthorized:
+            _LOGGING.error('Authentication failed')
+            self.motion_detection = None
+            return self.motion_detection
+
+        if response.status_code != requests.codes.ok:
+            # If we didn't receive 200, abort
+            _LOGGING.debug('Unable to fetch motion detection.')
+            self.motion_detection = None
+            return self.motion_detection
+
+        try:
+            tree = ET.fromstring(response.text)
+            nmsp = tree.tag.split('}')[0].strip('{')
+            self.namespace = nmsp if nmsp.startswith('http') else XML_NAMESPACE
+            ET.register_namespace("", self.namespace)
+            enabled = tree.find(self.element_query('enabled'))
+
+            if enabled is not None:
+                self._motion_detection_xml = tree
+            self.motion_detection = {'true': True, 'false': False}[enabled.text]
+            return self.motion_detection
+
+        except AttributeError as err:
+            _LOGGING.error('Entire response: %s', response.text)
+            _LOGGING.error('There was a problem: %s', err)
+            self.motion_detection = None
+            return self.motion_detection
+
+    def enable_motion_detection(self):
+        self._set_motion_detection(True)
+
+    def disable_motion_detection(self):
+        self._set_motion_detection(False)
+
+    def _set_motion_detection(self, enable):
+        url = '%s/MotionDetection/1' % self.root_url
+        self.get_motion_detection()
```

Comment:
What happens if we send a request to enable motion detection and it's already turned on?  Is it possible we can eliminate this call and rely on a 200 response from the camera to ensure we get the state we want?